include/openssl/hpke.h

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright 2019-2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/**
+ * @file
+ * APIs and data structures for HPKE (RFC9180).
+ */
+
+#ifndef OSSL_HPKE_H
+# define OSSL_HPKE_H
+# pragma once
+
+# include <openssl/ssl.h>
+
+# ifdef __cplusplus
+extern "C" {
+# endif
+
+/*
+ * The HPKE modes
+ */
+# define OSSL_HPKE_MODE_BASE              0 /**< Base mode  */
```

Comment:
If any of these defines are not used by the public API then they should not be in this file..There is an include/internal and a include/crypto.. If they are only used with hpke.c then they can go into the crypto/hpke

Reply:
These modes are all usable from the outside, so should be here. In practice, only base mode is used by applications so far.